app/screens/home/friends_screen.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect, useState } from "react";
+import { Text, StyleSheet } from "react-native";
+import { TopBar } from "@/components/navigation/TopBar";
+import { ThemedView } from "@/components/theme/ThemedView";
+import { ThemedText } from "@/components/theme/ThemedText";
+import { DBUser } from "@/firebase/FirestoreCtrl";
+import { getAuth } from "firebase/auth";
+import { SearchBar } from "@/components/friends/SearchBar";
+import ListOfFriends from "@/components/friends/ListOfFriends";
+import RequestList from "@/components/friends/RequestsList";
+import ListOfFilteredUsers from "@/components/friends/ListOfFilteredUsers";
+
+/**
+ * Screen that allowos the user to search for friends, add them and manage friend requests
+ * @param navigation : navigation prop
+ * @param firestoreCtrl : firestore controller
+ *
+ * @returns Friends Screen Component
+ */
+export default function FriendsScreen({ navigation, firestoreCtrl }: any) {
+  const [searchText, setSearchText] = useState("");
+  const [users, setUsers] = useState<DBUser[]>([]);
+  const [requests, setRequests] = useState<DBUser[]>([]);
+  const [friends, setFriends] = useState<DBUser[]>([]);
+
+  const auth = getAuth();
+  const uid = auth.currentUser?.uid;
+
+  // Fetch users
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const users = await firestoreCtrl.getAllUsers();
```

Comment:
This method is very in computationally expensive. Use instead the query I offered. This will then search for the user_name or contains the regex searched for.

Reply:
Same comment as before.